### PR TITLE
Add vendor stack

### PR DESCRIPTION
**Pull request by YuryHrytsuk**
## What do these changes do?
* Add vendor stack 
* Introduce `authenticated_platform_user` middleware
---
**Implementation insights**
* In case AUTH Server is down, the routers using `authenticated_platform_user` middleware in this case will return `500`
* The middleware `authenticated_platform_user` is available as long as OPS Traefik is up and running
* Vendor's manual service is only available via public network 

## Related issue/s
- closes https://github.com/ITISFoundation/osparc-ops-environments/issues/830

## Related PR/s
- https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/1016

## Checklist
- [X] I tested and it works
- [X] The Stack has been included in CI Workflow
- [X] Service has resource limits and reservations
- [X] Service has placement constraints or is global
- [X] Service is restartable
  - [X] The restart is zero-downtime
- [X] Service can migrate to another Node without manual intervention
- [ ] https://github.com/ITISFoundation/osparc-ops-environments/issues/853

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]
beautiful!

**Review by mguidon — APPROVED** [review]
Thanks. We need to discuss how and who is pushing the images to private dockerhub

File: services/vendors/Makefile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+STACK_NAME = $(notdir $(shell pwd))
+TEMP_COMPOSE=.stack.${STACK_NAME}.yaml
+REPO_BASE_DIR := $(shell git rev-parse --show-toplevel)
+
+# TARGETS --------------------------------------------------
+include ${REPO_BASE_DIR}/scripts/common.Makefile
+
+.PHONY: up ## Deploys registry stack
+up: ${TEMP_COMPOSE} .init .env
+	docker stack deploy --with-registry-auth --prune --compose-file $< ${STACK_NAME}
```
Comment (mrnicegyu11) [inline services/vendors/Makefile:14]: is the indentation right here?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -5,5 +5,19 @@
 ## Related PR/s
 
 ## Checklist
-
 - [ ] I tested and it works
+
```
Comment (mrnicegyu11) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: very nice!

**Review by mrnicegyu11 — APPROVED** [review]
lgtm, does this need changes in osparc-config?

File: services/vendors/Makefile (hunk 1 above)
Comment (YuryHrytsuk) [reply services/vendors/Makefile:14]: Yes, although it looks indeed weird in "github" diff

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (YuryHrytsuk) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: yeah, it helped me in this PR actually :)

**Pull request merged**